### PR TITLE
CLET-97: added callback implementation for Android

### PR DESCRIPTION
**Pull request by valllllll2000**
Task https://fidelapi.atlassian.net/browse/CLET-97

### CODE REVIEW

File: ios/Events/CardVerificationChoiceSelectedObserver.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import Foundation
+
+import Foundation
+import Fidel
+
+struct CardVerificationChoiceSelectedObserver: EventObserver {
+    func startObserving(_ onEvent: @escaping (NSDictionary) -> Void) {
+        Fidel.onCardVerificationChoiceSelected = {
+            let dictionary: NSMutableDictionary = NSMutableDictionary()
+            switch($0) {
```
Comment (valllllll2000) [inline ios/Events/CardVerificationChoiceSelectedObserver.swift:17]: TODO: add adapter

File: ios/Events/CardVerificationChoiceSelectedObserver.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            case .delegatedToThirdParty:
+                dictionary["verificationChoice"] = "ON_THE_SPOT"
```
Comment (cornelfd) [inline ios/Events/CardVerificationChoiceSelectedObserver.swift:19]: All the constants that we exposed, so far, use camel case (check the countries, for example). For consistency, I would suggest maintaining this convention.

File: ios/Events/CardVerificationChoiceSelectedObserver.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
+//
+//  CardVerificationChoiceSelectedObserver.swift
+//  fidel-react-native
+//
+//  Created by Valeria Rogatchevskikh on 8/6/23.
+//
+
+import Foundation
+
+import Foundation
```
[the remark this answers is not in the document]
Comment (valllllll2000) [reply ios/Events/CardVerificationChoiceSelectedObserver.swift:10]: oh that's an error, will remove

File: ios/Constants/CardVerificationChoiceConstants.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        case .onTheSpot: return "onTheSpot"
+        case .delegatedToThirdParty: return "delegatedToThirdParty"
+        @unknown default: return "unexpected"
+        }
+    }
+    
+    public static var allCases: Set<CardVerificationChoice> = [CardVerificationChoice.onTheSpot, .delegatedToThirdParty]
+    static var parentKeyName: String = String(describing: Self.self)
+    
+    static func verificationChoice(from constantKey: String?) -> CardVerificationChoice? {
```
Comment (cornelfd) [inline ios/Constants/CardVerificationChoiceConstants.swift:24]: Is this function used anywhere?

**Review by cornelfd — APPROVED** [review]

File: ios/Constants/CardVerificationChoiceConstants.swift (hunk 4 above)
Comment (valllllll2000) [reply ios/Constants/CardVerificationChoiceConstants.swift]: you were right, not used, so I removed it

**Pull request merged**